Gems/Atom/RHI/DX12/Code/Source/RHI/AsyncUploadQueue.cpp

```diff
@@ -512,39 +527,23 @@ namespace AZ
             }
         }
 
-        void AsyncUploadQueue::QueueTileMapping(CommandList::TileMapRequest& request)
+        void AsyncUploadQueue::QueueTileMapping(const CommandList::TileMapRequest& request)
         {
+            AZStd::scoped_lock lock{m_copyQueueMutex};
+            m_tileMapRequests.push(request);
+
+            const CommandList::TileMapRequest& cachedRequest = m_tileMapRequests.back();
```

Comment:
This is copying the request into a list just to keep it in memory while the lambda binds a reference to it, right?
Instead, could you just bind a copy in the lambda? 
(Same for QueueUpload and m_imageExpandRequests)
```
void AsyncUploadQueue::QueueTileMapping(const CommandList::TileMapRequest& request)
{
    CommandList::TileMapRequest cachedRequest = request;
    m_copyQueue->QueueCommand([cachedRequest ](void* commandQueue)
        {
            ID3D12CommandQueue* dx12CommandQueue = static_cast<ID3D12CommandQueue*>(commandQueue);
            UpdateTileMap(dx12CommandQueue, cachedRequest);
        });
}
```

Reply:
The UpdateTileMap also need to be changed from 
`UpdateTileMap(const CommandList::TileMapRequest& request)`
to 
`UpdateTileMap(CommandList::TileMapRequest request)`
to make it working. 

Although I didn't change it this way because it won't be consistent with all the other calls of CommandQueue which uses const & instead of object.

Reply:
You can still pass by reference to UpdateTileMap. In my suggestion, I'm copying into a local variable and then binding to the lambda by value. This will copy the value into the lambda instead of copying the reference.

Reply:
I think it would only work if I specify the TileMapRequest object in the lambda explicitly instead of using '='. That means I need to add all the individual variables in the lambda. This is fine for this function since its body is small and didn't use much of other variables from the class. But the lambda function in image expand function used a lot of members from the class which requires a long list of for the input.

Reply:
I'm pretty sure with just "[=]" you will capture everything by value, both the _this_ pointer and the local variables.